File: src/services/personal.service.ts
import api from './api';

// ── Personal Files ────────────────────────────────────────────────

export async function getPersonalFiles(userId: string) {
  return api.get('/personal/files', { params: { userId } });
}

export async function uploadPersonalFile(data: Record<string, any>) {
  return api.post('/personal/files', data);
}

export async function deletePersonalFile(id: string) {
  return api.delete(`/personal/files/${id}`);
}

// ── Notes ─────────────────────────────────────────────────────────

export async function getNotes(userId: string, params?: Record<string, any>) {
  return api.get('/personal/notes', { params: { userId, ...params } });
}

export async function createNote(data: Record<string, any>) {
  return api.post('/personal/notes', data);
}

export async function updateNote(id: string, data: Record<string, any>) {
  return api.patch(`/personal/notes/${id}`, data);
}

export async function deleteNote(id: string) {
  return api.delete(`/personal/notes/${id}`);
}

// ── Favorites ─────────────────────────────────────────────────────

export async function getFavorites(userId: string, params?: Record<string, any>) {
  return api.get('/personal/favorites', { params: { userId, ...params } });
}

export async function addFavorite(data: { userId: string; entityType: string; entityId: string }) {
  return api.post('/personal/favorites', data);
}

export async function removeFavorite(id: string) {
  return api.delete(`/personal/favorites/${id}`);
}

File: syncflow-java/syncflow-common/src/main/java/com/syncflow/common/service/personal/PersonalService.java
package com.syncflow.common.service.personal;

import com.syncflow.common.dto.personal.*;
import com.syncflow.common.result.PageResult;

import java.util.List;

/**
 * Personal workspace service interface (files + notes).
 */
public interface PersonalService {

    /**
     * Get personal files for the current user.
     */
    PageResult<PersonalFileVO> getPersonalFiles(int pageNum, int pageSize);

    /**
     * Create a personal file entry.
     */
    PersonalFileVO createPersonalFile(CreatePersonalFileDTO dto);

    /**
     * Delete a personal file.
     */
    void deletePersonalFile(Long id);

    /**
     * Get notes for the current user.
     */
    PageResult<NoteVO> getNotes(String keyword, int pageNum, int pageSize);

    /**
     * Create a note.
     */
    NoteVO createNote(CreateNoteDTO dto);

    /**
     * Update a note.
     */
    NoteVO updateNote(Long id, CreateNoteDTO dto);

    /**
     * Delete a note.
     */
    void deleteNote(Long id);
}
